### PR TITLE
Statusnotifier

### PR DESCRIPTION
**Pull request by ugilio**
KDE 5 systray support - See also notes on pull request #1243

### CODE REVIEW

**Comment by Alexey-Yakovenko** [comment]
Is it really necessary to have the versions for GTK2 and GTK3 separately? I assumed the plugin would be using only glib, gdkpixbuf and the dbus bindings.. Then it needs only 1 .so, which should be compatible with both GTK2 and 3.

**Comment by ugilio** [comment]
Ah... yes I think you are right. I think only pixbuf, glib and gio are needed and they are independent of gtk. I did not know (it's the first time I ever dive into glib/gtk).
I am going to change this, it simplifies things a bit

**Comment by Alexey-Yakovenko** [comment]
This PR is quite large. I think we'll have to do a few iterations on it before it lands. Hope it's ok with you. I want to get as little problems as possible when it's merged, so I will build and test it first, etc. But that will take a bit of time. I'll try to do time this weekend.

**Comment by ugilio** [comment]
Yes, no problem

**Comment by ugilio** [comment]
Ok, now there is only one plugin that can be used by both UIs
d2a482d60c0884a641e8797dfc8e11aaa061c5a1

File: plugins/statusnotifier/Makefile.am [hunk 1]
```diff
@@ -0,0 +1,23 @@
+if HAVE_STATUSNOTIFIER
+pkglib_LTLIBRARIES = statusnotifier.la
+
+AM_CFLAGS = $(CFLAGS) -std=c99 -fPIC
+
+statusnotifier_la_SOURCES = statusnotifier.c statusnotifier.h ddb_statusnotifier.c ddb_statusnotifier.h
+statusnotifier_la_LDFLAGS = -module -avoid-version
+# if STATICLINK
```
Comment (Alexey-Yakovenko) [inline plugins/statusnotifier/Makefile.am]: please remove all commented out lines, they're wrong anyway.

Comment (ugilio) [inline plugins/statusnotifier/Makefile.am]: done in b29cf83e6a2ee2021c3dbc542397a05671f264e3

File: plugins/statusnotifier/Makefile.am [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+AM_CFLAGS = $(CFLAGS) -std=c99 -fPIC
+
+statusnotifier_la_SOURCES = statusnotifier.c statusnotifier.h ddb_statusnotifier.c ddb_statusnotifier.h
+statusnotifier_la_LDFLAGS = -module -avoid-version
+statusnotifier_la_LIBADD = $(LDADD) $(STATUSNOTIFIER_DEPS_LIBS)
+if HAVE_GTK2
+statusnotifier_la_CFLAGS = -std=c99 $(GTK2_DEPS_CFLAGS)
+else
+statusnotifier_la_CFLAGS = -std=c99 $(GTK3_DEPS_CFLAGS)
```
Comment (Alexey-Yakovenko) [inline plugins/statusnotifier/Makefile.am]: plugins/statusnotifier/Makefile.am:18: statusnotifier_la_CFLAGS multiply defined in condition HAVE_STATUSNOTIFIER ...
plugins/statusnotifier/Makefile.am:16: ... `statusnotifier_la_CFLAGS' previously defined here

Comment (ugilio) [inline plugins/statusnotifier/Makefile.am]: this was fixed too in b29cf83e6a2ee2021c3dbc542397a05671f264e3

**Comment by Alexey-Yakovenko** [comment]
I'm merging this on the statusnotifier branch in my repo. I made a few modifications to remove the GTK dependency, and be compatible with both GTKUI versions correctly. Will post updates here.

**Comment by ugilio** [comment]
Ok

**Comment by Alexey-Yakovenko** [comment]
The branch is up now: https://github.com/Alexey-Yakovenko/deadbeef/tree/statusnotifier
We'll need to make sure it builds and works (including drone.io builds), then we'll be able to apply it to master branch.

**Comment by Alexey-Yakovenko** [comment]
The build has worked, please try it out: https://drone.io/github.com/Alexey-Yakovenko/deadbeef/files

**Comment by ugilio** [comment]
The builds do not include statusnotifier.so. Maybe the packing script needs to be updated. Also, I noticed this in the build logs:

make[2]: Entering directory `/home/ubuntu/src/github.com/Alexey-Yakovenko/deadbeef/plugins/statusnotifier'
  CC     statusnotifier_la-statusnotifier.lo
  CC     statusnotifier_la-ddb_statusnotifier.lo
statusnotifier.c: In function ‘sn_seticondata’:
statusnotifier.c:201:5: warning: implicit declaration of function ‘gdk_pixbuf_get_byte_length’ [-Wimplicit-function-declaration]
statusnotifier.c: In function ‘icondata_to_variant’:
statusnotifier.c:549:9: warning: implicit declaration of function ‘g_variant_new_fixed_array’ [-Wimplicit-function-declaration]
statusnotifier.c:549:29: warning: initialization makes pointer from integer without a cast [enabled by default]
  CCLD   statusnotifier.la
make[2]: Leaving directory `/home/ubuntu/src/github.com/Alexey-Yakovenko/deadbeef/plugins/statusnotifier'

They are new apis. I have a workaround for gdk_pixbuf_get_byte_length, but I have to check g_variant_new_fixed_array.

**Comment by Alexey-Yakovenko** [comment]
> They are new apis. I have a workaround for gdk_pixbuf_get_byte_length, but I have to check g_variant_new_fixed_array.

currently, the only simple way to fix the warnings is to add forward declarations for both functions.

**Comment by ugilio** [comment]
I just updated the code. I am opening a new PR against your statusnotifier branch

**Pull request closed without merging**